core-bundle/src/Routing/Matcher/DomainFilter.php

```diff
@@ -42,7 +42,9 @@ public function filter(RouteCollection $collection, Request $request): RouteColl
 
         if ($hasDomainMatch) {
             foreach ($collection->all() as $name => $route) {
-                if (!$route->getHost()) {
+                $host = $route->getHost();
+
+                if (!$host || $host !== $httpHost) {
```

Comment:
@aschempp I am not 100% sure if this is right, but filtering routes which have a domain that does not match the host (instead of just the ones that have no domain at all) does decrease the number of potential matches.

Reply:
Yeah, I think this makes sense. I just didn't add it because they would not match anyway. But this should theoretically result in a smaller number of route match tests.